### PR TITLE
Add Github action to generate prerelease builds

### PR DESCRIPTION
**Pull request by manodnyab**
## Description

This generates prerelease builds for all 3 plugins

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by justinmk3 — APPROVED** [review]
Nice

File: .github/workflows/prerelease.yml [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-java@v4
+        with:
+          distribution: 'corretto'
+          java-version: '17'
+      - uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: '6.x'
```
Comment (rli) [inline .github/workflows/prerelease.yml:82]: core an q don't need dotnet

File: .github/workflows/prerelease.yml [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+      - name: Generate artifact
+        run: |
+          ./gradlew -PideProfileName=${{ matrix.supported_versions }} :plugin-toolkit:intellij:buildPlugin --exclude-task buildResharperPlugin
```
Comment (rli) [inline .github/workflows/prerelease.yml:33]: let's also do `intellij-standalone`

File: .github/workflows/prerelease.yml [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+      - name: Upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: plugin-toolkit-${{ matrix.supported_versions }}
+          path: ./plugins/toolkit/intellij/build/distributions/*.zip
+          retention-days: 1
+
+  generate_artifact_q:
```
Comment (rli) [inline .github/workflows/prerelease.yml:41]: as a future optimization, we can combine these all into a single job and just run `./gradlew buildPlugin`
as-is we're repeatedly downloading tens of gigs of dependencies

we will need to disable that task on the subprojects where it does not make sense

Comment (manodnyab) [reply .github/workflows/prerelease.yml:40]: Yeah, that sounds good, wanted to keep this separate currently cause:
1) Their creation is independent of each other 
2) the running out of disk space issue

File: .github/workflows/prerelease.yml (hunk 2 above)
Comment (manodnyab) [reply .github/workflows/prerelease.yml:32]: Added

**Review by rli — APPROVED** [review]

**Pull request merged**